miio/tests/test_device.py

```diff
@@ -59,6 +59,18 @@ def test_unavailable_device_info_raises(mocker):
     assert send.call_count == 1
 
 
+def test_device_id_handshake(mocker):
+    """Make sure send_handshake() gets called if did is unknown."""
```

Comment:
This looks good to me, please also add a test for the code path where the did is known.

Reply:
I just added the second test